core/amber/src/main/resources/python_udf/texera_udf_server_main.py

```diff
@@ -159,19 +172,67 @@ def _output_data(self):
             output_data_list.append(self.udf_op.next())
         output_dataframe = pandas.DataFrame.from_records(output_data_list)
         # send output data to Java
-        output_key = self._descriptor_to_key(self._accept(b'fromPython'))
+        self._send_flight("fromPython", output_dataframe)
+
+    def _get_flight(self, channel: str) -> pandas.DataFrame:
+        self.logger.debug(f"transforming flight {channel.__repr__()}")
+        df = self.flights[self._descriptor_to_key(self._to_descriptor(channel.encode()))].to_pandas()
+        self.logger.debug(f"got {len(df)} rows in this flight")
+        return df
+
+    def _remove_flight(self, channel: str) -> None:
+        self.logger.debug(f"removing flight {channel.__repr__()}")
+        self.flights.pop(self._descriptor_to_key(self._to_descriptor(channel.encode())))
+
+    def _send_flight(self, channel: str, output_dataframe: pandas.DataFrame) -> None:
+        output_key = self._descriptor_to_key(self._to_descriptor(channel.encode()))
+        self.logger.debug(f"prepared {len(output_dataframe)} rows in this flight")
+        self.logger.debug(f"sending flight {channel.__repr__()}")
         self.flights[output_key] = pyarrow.Table.from_pandas(output_dataframe)
 
     @staticmethod
     def _response(message: bytes):
         return pyarrow.flight.Result(pyarrow.py_buffer(message))
 
     @staticmethod
-    def _accept(channel: bytes):
+    def _to_descriptor(channel: bytes) -> FlightDescriptor:
         return pyarrow.flight.FlightDescriptor.for_path(channel)
 
+    def _configure(self, *args):
+        self._setup_logger(*args)
+
+    def _setup_logger(self, *args):
+        # TODO: make it kwargs
+        # create a file handler
+
+        log_dir = args[0]
+        file_name = f"{datetime.utcnow().isoformat()}-{os.getpid()}"
```

Comment:
please change

Reply:
changed to `texera-python_udf-{time}-{pid}.log`